modules/button/react/lib/ButtonStyles.ts

```diff
@@ -311,7 +312,8 @@ export const iconButtonStyles: ButtonGenericStyle = {
   variants: {
     sizes: {
       [ButtonSize.Small]: {
-        width: canvas.spacing.l,
+        minWidth: canvas.spacing.l,
+        width: 'auto',
```

Comment:
The small buttons are normally of fixed size. This gives them a minimum width and allows their width to grow based on the content. In my case, this was very helpful for numbers like `10,000`

Reply:
@lychyi @NicholasBoll just want to confirm you are okay with this change

Reply:
It's okay with me

Reply:
It may be a good idea to add some visual stories to ensure that they work properly in all browsers. I know `width` can be tricky across browsers.

Reply:
Each story handles resize overrides. Can we add chromatic directives instead?